server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -461,6 +516,7 @@ public void invariant() {
                 assert leaderChecker.currentNodeIsMaster() == false;
                 assert leaderCheckScheduler != null;
                 assert followersChecker.getKnownFollowers().isEmpty();
+                assert currentPublication.map(Publication::isCommitted).orElse(true);
```

Comment:
Nice. I think this is a strong enough link between `currentPublication` and `mode`.